python/dgl/backend/tensorflow/tensor.py

```diff
@@ -0,0 +1,564 @@
+    return tf.transpose(input, perm=[axis1, axis2])
+
+
+def zeros(shape, dtype, ctx):
+    with tf.device(ctx):
+        t = tf.zeros(shape, dtype=dtype)
+    return t
+
+
+def zeros_like(input):
+    return tf.zeros_like(input)
+
+
+def ones(shape, dtype, ctx):
+    with tf.device(ctx):
+        t = tf.ones(shape, dtype=dtype)
+    return t
+
+
+def uniform(shape, dtype, ctx, low, high):
+    with tf.device(ctx):
+        t = tf.random.uniform(shape, dtype=dtype, minval=low, maxval=high)
+    return t
+
+
+def pad_packed_tensor(input, lengths, value, l_min=None):
+    old_shape = input.shape
+    if isinstance(lengths, tf.Tensor):
+        max_len = as_scalar(lengths.max())
+    else:
+        max_len = builtins.max(lengths)
+
+    if l_min is not None:
+        max_len = builtins.max(max_len, l_min)
+
+    batch_size = len(lengths)
+    ndim = input.ndim
+    tensor_list = []
+    cum_row = 0
+    pad_nparray = np.zeros((ndim, 2), dtype=np.int32)
+    for l in lengths:
+        t = input[cum_row:cum_row+l]
+        pad_nparray[0, 1] = max_len - l
+        t = tf.pad(t, tf.constant(pad_nparray),
+                   mode='CONSTANT', constant_values=value)
+        tensor_list.append(t)
+        cum_row += l
+    return tf.stack(tensor_list, axis=0)
+
+
+def pack_padded_tensor(input, lengths):
+    out_list = []
+    for i, l in enumerate(lengths):
+        t = input[i]
+        out = t[:l]
+        out_list.append(out)
+    return tf.concat(out_list, axis=0)
+
+
+def unsorted_1d_segment_sum(input, seg_id, n_segs, dim):
+    assert dim == 0  # Why we need dim for 1d?
+    return tf.math.unsorted_segment_sum(input, seg_id, n_segs)
+
+
+def unsorted_1d_segment_mean(input, seg_id, n_segs, dim):
+    assert dim == 0  # Why we need dim for 1d?
+    return tf.math.unsorted_segment_mean(input, seg_id, n_segs)
+
+# TODO: TF has unsorted_segment_max, which can accelerate _max_on on batched graph
+
+
+def boolean_mask(input, mask):
+    return tf.boolean_mask(input, mask)
+
+
+def equal(x, y):
+    return x == y
+
+
+def logical_not(input):
+    return ~input
+
+
+def unique(input):
+    return tf.unique(input).y
+
+
+def full_1d(length, fill_value, dtype, ctx):
+    with tf.device(ctx):
+        t = tf.fill([length], value=fill_value)
+        t = tf.cast(t, dtype=dtype)
+    return t
+
+
+def nonzero_1d(input):
+    nonzero_bool = (input != False)
+    return tf.reshape(tf.where(nonzero_bool), (-1, ))
+
+
+def sort_1d(input):
+    return tf.sort(input), tf.cast(tf.argsort(input), dtype=tf.int64)
+
+
+def arange(start, stop, ctx=None):
+    if ctx is None:
+        return tf.range(start, stop, dtype=tf.int64)
+    else:
+        with tf.device(ctx):
+            t = tf.range(start, stop, dtype=tf.int64)
+        return t
+
+
+def rand_shuffle(arr):
+    return tf.random.shuffle(arr)
+
+
+def zerocopy_to_dlpack(input):
+    return tfdlpack.to_dlpack(input)
+
+
+def zerocopy_from_dlpack(dlpack_tensor):
+    # Not zero copy
```

Comment:
Is it?

Reply:
Yes....I forgot to clean the comment here.